test/metabase/notification/payload/temp_storage_test.clj

```diff
@@ -0,0 +1,30 @@
+(ns metabase.notification.payload.temp-storage-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.notification.payload.temp-storage :as temp-storage]))
+
+(set! *warn-on-reflection* true)
+
+(deftest to-temp-file!-test
+  (testing "basic storage and retrieval works"
+    (let [data {:test "data"}
+          stored (temp-storage/to-temp-file! data)]
+      (is (= data @stored))))
+
+  (testing "can store and retrieve complex data structures"
+    (let [data {:a [1 2 3]
+                :b #{"set" "of" "strings"}
+                :c {:nested "map"}}
+          stored (temp-storage/to-temp-file! data)]
+      (is (= data @stored))))
+
+  (testing "file is deleted after specified timeout"
+    (let [data {:test "temporary"}
+          stored (temp-storage/to-temp-file! data 1)]
+      (is (= data @stored))
+      (Thread/sleep 1500)
```

Comment:
If you have a constant, you can avoid this coupled magic number.

Reply:
are you referring to the constant for default lifetime ? if so the "1500" constant here is relative to the "1" second we're using for this test, so I think it's still needed?

Reply:
Ah, missed that you're not using the default here 👍